c7n/resources/vpc.py

```diff
@@ -1094,6 +1096,24 @@ def process_self_reference(self, perm, sg_id):
                 found = True
         return found
 
+    def process_sg_references(self, perm):
+        sg_refs = self.data.get('SGReferences')
+        if sg_refs:
+            sg_perm = perm.get('UserIdGroupPairs', [])
```

Comment:
this needs to be checking for cross account sg refs and elliding/removing them, as there won't be access to check fetch them.


also this entire method body can be dendented  by if not sg_refs: return